### ISSUE TITLE
[typedef] broken for constructor argument with access modificator and default value

### ISSUE THREAD

**Issue by mrmlnc** [typescript-eslint/typescript-eslint#921]
**Related issues**

* #784 
* #785

**Repro**

```json
{
    "parser": "@typescript-eslint/parser",
    "plugins": [
        "@typescript-eslint"
    ],
    "rules": {
        "@typescript-eslint/typedef": [
            "error",
            {
                "parameter": true
            }
        ]
    }
}
```

```ts
type Options = {
    force?: boolean;
}

class Something {
    // Works fine without `= {}`
    constructor(private readonly _options: Options = {}) {
        // something
    }
}
```

**Expected Result**

All fine. The output has no errors.

**Actual Result**

```
…\typescript-eslint-repro\src\index.ts
  6:17  error  expected a type annotation  @typescript-eslint/typedef

✖ 1 problem (1 error, 0 warnings)
```

**Additional Info**

The `--debug` argument could tell me nothing of interest.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.0.0` |
| `@typescript-eslint/parser`        | `2.0.0` |
| `TypeScript`                       | `3.5.3` |
| `ESLint`                           | `6.2.2` |
| `node`                             | `12.8.0` |
| `npm`                              | `6.10.2` |

**Comment by mrmlnc** [comment]
Fixed by #923.
